File: AED 2/C Programs/Trabalho Prático/q4.c
#include <stdio.h>
#include <string.h>

//declaração de função
void criptografaString(char v[], int i, int len);

int main(void) {
  char v[1000], aux;

  // rodar ate encontrar string FIM
  scanf(" %[^\n]s", v);
  while (v[0] != 'F' && v[1] != 'I' && v[2] != 'M') {
    criptografaString(v, 0, strlen(v));
    scanf(" %[^\n]s", v);
  }

  return 0;
}

// função recusrsiva para criptografar letra por letra em cada chamada
void criptografaString(char v[], int i, int len) {
  if (i < len) {
    // verifica se chegou ao final da string atraves da palavra FIM
    if (v[i] == 'F' && v[i + 1] == 'I' && v[i + 2] == 'M') {
      return;
    }
    // verifica se caracter esta contitido no alfabeto de acordo com a tabela ascii (imprimir espacos vazios)
    if ((v[i] >= 65 && v[i] <= 87) || (v[i] >= 97 && v[i] <= 119))
      // adiciona 3 numeros a letra caso nao seja uma das 3 ultimas
      printf("%c", v[i] + 3);
    else {
      printf(" ");
    }

    // verifica se as letras sao as 3 ultimas do alfabeto
    if ((v[i] >= 120 && v[i] <= 122) || (v[i] >= 88 && v[i] <= 90)) {
      printf("%c", v[i] - 23);
    }

    // chama a função novamente para a proxima letra ser criptografada
    criptografaString(v, ++i, len);
  }
}

File: AED 2/Java Programs/src/Trabalho_Pratico_1/q4.java
package Trabalho_Pratico_1;

import java.util.Scanner;

public class q4 {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    String v;

    // leitura de string
    v = in.nextLine();
    in.close();
    criptografaString(v, 0);
  }

  // metodo recusrsivo para criptografar letra por letra em cada chamada
  public static void criptografaString(String v, int i) {
    if (i < v.length()) {
      // verifica se chegou ao final da string atraves da palavra FIM
      if (v.charAt(i) == 'F' && v.charAt(i + 1) == 'I' && v.charAt(i + 2) == 'M') {
        return;
      }
      // verifica se caracter esta contitido no alfabeto de acordo com a tabela ascii
      // (imprimir espacos vazios)
      if ((v.charAt(i) >= 65 && v.charAt(i) <= 87) || (v.charAt(i) >= 97 && v.charAt(i) <= 119))
        // adiciona 3 numeros a letra caso nao seja uma das 3 ultimas
        System.out.printf("%c", v.charAt(i) + 3);
      else {
        System.out.printf(" ");
      }

      // verifica se as letras sao as 3 ultimas do alfabeto
      if ((v.charAt(i) >= 120 && v.charAt(i) <= 122) || (v.charAt(i) >= 88 && v.charAt(i) <= 90)) {
        if (v.charAt(i) == 'X') {
          System.out.printf("A");
        } else if (v.charAt(i) == 'x') {
          System.out.printf("a");
        } else if (v.charAt(i) == 'Y') {
          System.out.printf("B");
        } else if (v.charAt(i) == 'y') {
          System.out.printf("b");
        } else if (v.charAt(i) == 'Z') {
          System.out.printf("B");
        } else if (v.charAt(i) == 'z') {
          System.out.printf("b");
        }
      }
      // chama a função novamente para a proxima letra ser criptografada
      criptografaString(v, ++i);
    }
  }
}
